src/main/java/pl/mmieczak/springbootexercise_22_1/UserController.java

```diff
@@ -44,5 +32,39 @@ public String showList() {
         String tableEnd = "</tbody></table>";
         return tableBegin + itemsList + tableEnd;
     }
-}
 
+    @ResponseBody
+    @RequestMapping(value = "/submit", method = RequestMethod.POST, params = {"search"})
+    public String search(@RequestParam(defaultValue = "", required = true) String name, @RequestParam(defaultValue = "", required = true) String surname, @RequestParam(defaultValue = "-1", required = false) String age) {
+        Stream<User> filteredData = userRepository.filterDataUsingFormValues(name, surname, age);
```

Comment:
Stream jako typ zwracany jest bardzo kontrowersyjny. Część programistów (duża część) twierdzi że nie powinniśmy tego robić, z racji tego że Stream może być w stanie zużytym albo nowym (który można wykorzystać). Lepiej jest zawsze zwrócić kolekcję i na niej wykonać `.stream()`

Reply:
Ciekawe spostrzeżenie nie sądziłem ze to. może być błąd, dzięki :)